lib/rails/generators/haml/component_generator.rb

```diff
@@ -7,9 +7,12 @@ module Generators
     class ComponentGenerator < Erb::Generators::ComponentGenerator
```

Comment:
I'm wondering why this generator inherits from `Erb::Generators::ComponentGenerator` instead of `Base` like [there](https://github.com/github/view_component/blob/main/lib/rails/generators/erb/component_generator.rb#L7)?

Reply:
I found this relevant discussion: https://github.com/github/view_component/pull/190#pullrequestreview-345518807 but maybe it's time to reconsider this?

Reply:
I suspect this still comes down to how much do we want to future proof an upstream into Rails.